central/complianceoperator/v2/benchmarks/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,195 @@
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.ComplianceOperator)))
+	s.noAccessCtx = sac.WithGlobalAccessScopeChecker(context.Background(), sac.DenyAllAccessScopeChecker())
+
+	s.mockCtrl = gomock.NewController(s.T())
+
+	s.db = pgtest.ForT(s.T())
+	s.storage = benchmarkStorage.New(s.db)
+	s.datastore = New(s.storage)
+}
+
+func (s *complianceBenchmarkDataStoreSuite) TearDownTest() {
+	s.db.Teardown(s.T())
+}
+
+func (s *complianceBenchmarkDataStoreSuite) TestUpsertBenchmark() {
+	// make sure we have nothing
+	benchmarkIDs, err := s.storage.GetIDs(s.hasReadCtx)
+	s.Require().NoError(err)
+	s.Require().Empty(benchmarkIDs)
+
+	b1 := getTestBenchmark(uuidStub1, "b1", "1.0", 1)
+	b2 := getTestBenchmark(uuidStub2, "b2", "2.0", 2)
+
+	s.Assert().NoError(s.datastore.UpsertBenchmark(s.hasWriteCtx, b1))
+	s.Assert().NoError(s.datastore.UpsertBenchmark(s.hasWriteCtx, b2))
+
+	count, err := s.storage.Count(s.hasReadCtx, search.EmptyQuery())
+	s.Assert().NoError(err)
+	s.Assert().Equal(2, count)
+
+	s.Assert().Error(s.datastore.UpsertBenchmark(s.hasReadCtx, b1))
+
+	retB1, found, err := s.storage.Get(s.hasReadCtx, b1.GetId())
+	s.Assert().NoError(err)
+	s.Assert().True(found)
+	assertBenchmarks(s.T(), b1, retB1)
+
+	retB2, found, err := s.storage.Get(s.hasReadCtx, b2.GetId())
+	s.Assert().NoError(err)
+	s.Assert().True(found)
+	assertBenchmarks(s.T(), b2, retB2)
+}
+
+func (s *complianceBenchmarkDataStoreSuite) TestDeleteBenchmark() {
+	// make sure we have nothing
+	benchmarkIDs, err := s.storage.GetIDs(s.hasReadCtx)
+	s.Require().NoError(err)
+	s.Require().Empty(benchmarkIDs)
+
+	b1 := getTestBenchmark(uuidStub1, "b1", "1.0", 1)
+	b2 := getTestBenchmark(uuidStub2, "b2", "2.0", 2)
+
+	s.Require().NoError(s.storage.Upsert(s.hasWriteCtx, b1))
+	s.Require().NoError(s.storage.Upsert(s.hasWriteCtx, b2))
+
+	count, err := s.storage.Count(s.hasReadCtx, search.EmptyQuery())
+	s.Require().NoError(err)
+	s.Require().Equal(2, count)
+
+	s.Assert().NoError(s.datastore.DeleteBenchmark(s.hasWriteCtx, b1.GetId()))
+
+	count, err = s.storage.Count(s.hasReadCtx, search.EmptyQuery())
+	s.Assert().NoError(err)
+	s.Assert().Equal(1, count)
+
+	s.Assert().Error(s.datastore.DeleteBenchmark(s.hasReadCtx, b2.GetId()))
+
+	s.Assert().NoError(s.datastore.DeleteBenchmark(s.hasWriteCtx, b2.GetId()))
+
+	count, err = s.storage.Count(s.hasReadCtx, search.EmptyQuery())
+	s.Assert().NoError(err)
+	s.Assert().Equal(0, count)
+}
+
+func (s *complianceBenchmarkDataStoreSuite) TestGetBenchmark() {
+	// make sure we have nothing
+	benchmarkIDs, err := s.storage.GetIDs(s.hasReadCtx)
+	s.Require().NoError(err)
+	s.Require().Empty(benchmarkIDs)
+
+	benchmarks := []*storage.ComplianceOperatorBenchmarkV2{
+		getTestBenchmark(uuidStub1, "b1", "1.0", 1),
+		getTestBenchmark(uuidStub2, "b2", "2.0", 2),
+	}
+
+	for _, b := range benchmarks {
```

Comment:
assert slice length before for loops

Reply:
I'm not sure if this is needed here. The `benchmarks` slice is hardcoded in the test.